File: November-18-2015/Solutions/Solution2.rb
def translate(word)
  change = false

  if word == word.capitalize
    word.downcase!
    change = true
  end

  if (word[0] =~ /[aeiou]/)
    word = word + 'way'
  elsif word.include?('qu') && (word.index('qu') == 0)
    word = word[2..-1].concat('quay')
  else
    idx = word.index(/[aeiou]/)
    word = word[idx..-1].concat(word[0...idx]).concat('ay')
  end

  word.capitalize! if change
  return word
end





def pig_latin(str)
  result = str.split(" ").map { |word|
    translate(word)
    }.join(" ")
  result
end


p pig_latin("Quiet tree") =="Ietquay eetray"
p pig_latin("apple quick Fox") == "appleway ickquay Oxfay"

File: November-18-2015/Solutions/Solution2.js
function pig_latin_translator(words) {
	var vowel = "aeiou",
	    inputArr = words.split(" "),
	    resultArray = [],
	    index = 0,
	    start,
	    end;

	inputArr.forEach(function(word) {
		for (var i = 0; i < word.length; i++) {
			if (vowel.indexOf(word[i]) !== -1) {
				if (word[i] != "u" || (word[i] === "u" && word[i-1] !== "q")) {
					index = i;
					break;
				}
			}
		}
		
		start = word.substring(index, word.length);
		end = word.substring(0, index) + "ay";
		
		resultArray.push( start + end );
	});

	return resultArray.join(" ");
}

pig_latin_translator("the quick brown fox"); // return 'ethay ickquay ownbray oxfay'